### ISSUE TITLE
Support for basic auth

### ISSUE THREAD

**Issue by pyfreak** [Digital-Sapphire/PyUpdater#69]
Would be nice if support for basic auth could be added to PyUpdater possibly add the settings into client_config.py? I managed to get this working by monkey patching get_http_pool. Unless I have completely missed something from the documentation 😃 

```
from client_config import ClientConfig
import pyupdater.client.downloader
from pyupdater.client import Client
import urllib3
import certifi

# Monkey patch pyupdater to use basic auth
def get_http_pool_monkey(secure=True):

    headers = urllib3.util.make_headers(basic_auth='%s:%s' % ("username","password"))

    if secure is True:
        http = urllib3.PoolManager(cert_reqs=str('CERT_REQUIRED'),
                                   ca_certs=certifi.where())
        http.headers.update(headers)
        return http
    else:
        http = urllib3.PoolManager()
        http.headers.update(headers)
        return http


pyupdater.client.downloader.get_http_pool = get_http_pool_monkey
```

**Comment by JMSwag** [comment]
@pyfreak Basic auth support would be a great addition! Your patch, 👍 , is essentially the implementation. A PR is welcome. If you don't have the time I'll add it to my to-do list.

😄

**Comment by JMSwag** [comment]
@pyfreak This is being worked on [here](https://github.com/JMSwag/PyUpdater/tree/issue-69).

Example below.

```
client = Client(ClientConfig(), headers={'basic_auth': 'user:passwd'})
```

**Comment by JMSwag** [comment]
@pyfreak Implemented in 71877b24e1fa69911612771b0641b3e341635dab. It'll be in [v2.5.0](https://github.com/JMSwag/PyUpdater/blob/master/docs/changelog.md).

**Comment by wolfamongus2** [comment]
@pyfreak I don't suppose you could show working example, this is not working for me.  I need to include basic auth + headers for bitbucket.
